tests/integration/test_types_integration.py

```diff
@@ -0,0 +1,278 @@
+        .add_field(sql="true", python=True) \
+        .add_field(sql="cast(null AS BOOLEAN)", python=None) \
+        .execute()
+
+
+def test_tinyint_types(trino_connection):
+    SqlTest(trino_connection) \
+        .add_field(sql="cast(127 AS TINYINT)", python=127) \
+        .add_field(sql="cast(-128 AS TINYINT)", python=-128) \
+        .add_field(sql="cast(42 AS TINYINT)", python=42) \
+        .add_field(sql="cast(-42 AS TINYINT)", python=-42) \
+        .add_field(sql="cast(null AS TINYINT)", python=None) \
+        .execute()
+
+
+def test_smallint_types(trino_connection):
+    SqlTest(trino_connection) \
+        .add_field(sql="cast(32767 AS SMALLINT)", python=32767) \
+        .add_field(sql="cast(-32768 AS SMALLINT)", python=-32768) \
+        .add_field(sql="cast(null AS SMALLINT)", python=None) \
+        .execute()
+
+
+def test_int_types(trino_connection):
+    SqlTest(trino_connection) \
+        .add_field(sql="cast(2147483647 AS INTEGER)", python=2147483647) \
+        .add_field(sql="cast(-2147483648 AS INTEGER)", python=-2147483648) \
+        .add_field(sql="cast(null AS INTEGER)", python=None) \
+        .add_field(sql="cast(9223372036854775807 AS BIGINT)", python=9223372036854775807) \
+        .add_field(sql="cast(-9223372036854775808 AS BIGINT)", python=-9223372036854775808) \
+        .add_field(sql="cast(null AS BIGINT)", python=None) \
+        .execute()
+
+
+def test_bigint_types(trino_connection):
+    SqlTest(trino_connection) \
+        .add_field(sql="cast(9223372036854775807 AS BIGINT)", python=9223372036854775807) \
+        .add_field(sql="cast(-9223372036854775808 AS BIGINT)", python=-9223372036854775808) \
+        .add_field(sql="cast(null AS BIGINT)", python=None) \
+        .execute()
+
+
+def test_real_types(trino_connection):
+    SqlTest(trino_connection) \
+        .add_field(sql="cast(3.4028235E38 AS REAL)", python=3.4028235e+38) \
+        .add_field(sql="cast(1.4E-45 AS REAL)", python=1.4e-45) \
+        .add_field(sql="cast('Infinity' AS REAL)", python=math.inf) \
+        .add_field(sql="cast('-Infinity' AS REAL)", python=-math.inf) \
+        .add_field(sql="cast('NaN' AS REAL)", python=math.nan) \
+        .add_field(sql="cast(null AS REAL)", python=None) \
+        .execute()
+
+
+def test_double_types(trino_connection):
+    SqlTest(trino_connection) \
+        .add_field(sql="cast(1.7976931348623157E308 AS DOUBLE)", python=1.7976931348623157e+308) \
+        .add_field(sql="cast(4.9E-324 AS DOUBLE)", python=5e-324) \
+        .add_field(sql="cast('Infinity' AS DOUBLE)", python=math.inf) \
+        .add_field(sql="cast('-Infinity' AS DOUBLE)", python=-math.inf) \
+        .add_field(sql="cast('NaN' AS DOUBLE)", python=math.nan) \
+        .add_field(sql="cast(null AS DOUBLE)", python=None) \
+        .execute()
+
+
+def test_decimal_types(trino_connection):
+    SqlTest(trino_connection) \
+        .add_field(sql="DECIMAL '10.3'", python=Decimal('10.3')) \
```

Comment:
In general it's easier to read such type based tests if they follow some consistent ordering.

e.g. `null` -> min value -> some random value -> max value.

Also for parametrized types like DECIMAL, TIMESTAMP it helps to keep types with same parameters together. i.e. all `DECIMAL(18,18)` are together, then others etc. etc.

Reply:
Done